hugo/content/guides/code-bundling.md

```diff
@@ -14,34 +14,44 @@ We generally recommend using [esbuild](https://esbuild.github.io/) to bundle Lan
 npm i --save-dev esbuild
 ```
 
-You can see a minimal configuration file below that bundles both your language server and your extension. It will run as a simple `node` script.
+You can see a minimal configuration file below that bundles both your language server and your extension.
+
+```ts
+//@ts-check
+import * as esbuild from 'esbuild';
 
-```js
 const watch = process.argv.includes('--watch');
-const minify = process.argv.includes('--minify'); // Use this flag for production usage
-const success = watch ? 'Watch build succeeded' : 'Build succeeded';
+const minify = process.argv.includes('--minify');
 
-require('esbuild').build({
-    // Two entry points, one for the extension, one for the language server
-    entryPoints: ['src/extension.ts', 'src/language-server/main.ts'],
-    outdir: 'out', // All bundles are put into this directory
-    bundle: true, // We want to create bundles for the extension and language server
-    external: ['vscode'], // the vscode-module is created on-the-fly during runtime and must be excluded
+const ctx = await esbuild.context({
+    entryPoints: ['src/extension.ts', 'src/language/main.ts'],
+    outdir: 'out',
+    bundle: true,
+    target: "es6",
+    loader: { '.ts': 'ts' },
+    external: ['vscode'], // the vscode-module is created on-the-fly and must be excluded.
     platform: 'node', // VSCode extensions run in a node process
-    sourcemap: !minify, // Sourcemaps help us debug our original TypeScript code even after bundling
-    watch: watch ? {
-        onRebuild(error) {
-            if (error) console.error('Watch build failed')
-            else console.log(success)
-        }
-    } : false,
+    sourcemap: !minify,
     minify
-})
-    .then(() => console.log(success))
-    .catch(() => process.exit(1));
+});
+
+if (watch) {
+    await ctx.watch();
+} else {
+    await ctx.rebuild();
+    ctx.dispose();
```

Comment:
BTW why is `dispose` only called in this case?

Reply:
The documentation actually says to do it for both cases, but for some reason it makes the `ctx.watch` call stop, and I have no idea why. Given that `watch` runs anyway until the process ends, there aren't any negative consequences of not calling `dispose`, so I'd guess it's fine that way.